### DISCUSSION TITLE
Power BI visual for LineUp with event data

### DISCUSSION THREAD

**Discussion by jzethofer** [lineupjs/lineupjs#641] [General]
Hi @thinkh @sgratzl ,

I am developing a Power BI visual with LineUp and a custom event data column. I wanted to ask if you are ok with me publishing this on the Power BI AppSource. @mstreit suggested, that the event data column might also be a beneficial addition for LineUp as a supported column type. 

Here is a short demo video:

https://github.com/lineupjs/lineupjs/assets/98895743/7f7c223d-e0e7-42ba-9909-d74a83cc3559

## Fetures:
 - **LineUp Data Table**
   - displaying categorical, numerical, or temporal data in LineUp.js
 - **Custom event data column**
   - Show events for each data row on a time scale
   - Interactive time scale: Zoom and drag functionality
   - Event types legend
   - Box plot: Visualize the spread of a future event based on similar observed data
   - Events in LineUp overview mode: tiny rectangles provide a compact view of events
   - Event reference column for comparing events that occured at different points in time
   - Heatmap summary visualization to compare distribution of events
   - Tooltips
   - Settings for event column
     - Event colors
     - Sorting by events and box plot values
     - Toggle which events and box plot values are displayed in normal and overview mode
     - Setting reference event for box plot
 - Filtering other visuals based on similar data IDs of selected rows
 - Filtering other visuals based on selected row ID
 - Save button for keeping LineUp state (filters, sorting, column order, ...) between Power BI reloads
 - Power BI Settings 
   - LineUp aggregation strategy
   - Event scale unit
   - Summary heatmap resolution
   - Data table name of similar data to be filtered

**Comment by mstreit** [comment]
@thinkh @sgratzl @puehringer, what do you think? Could this be a valuable column to be added to the default set (independent of PowerBI)? Event data are also relevant in the bio-medical domain.
/CC @dg-datavisyn

**Reply by thinkh** [reply]
@puehringer and me would like to have this feature integrated in LineUp, even though we don't have a concrete use case yet.

@jzethofer Please file a PR in the https://github.com/lineupjs/lineupjs repo. If you need any support please let us know. Thanks a lot for your work.
